.github/workflows/build.yml

```diff
@@ -23,35 +23,66 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: yarn
-      - run: yarn
+      - run: yarn install --ignore-scripts
+      - run: yarn prebuild
+      - run: yarn build:cjs
       - run: tar -czf /tmp/web3-${{ matrix.node }}.js.tar.gz --exclude="./.git" ./
       - uses: actions/upload-artifact@v3
         with:
           name: web3-${{ matrix.node }}.js.tar.gz
           path: /tmp/web3-${{ matrix.node }}.js.tar.gz
-
+  build-cjs-types:
+      name: Build ESM & Types
+      needs: build
```

Comment:
there is no need of `needs: build`  so it should be removed? as ESM is not dependent on CJS technical

Reply:
done

Reply:
@avkos It should though, our code is written using modules, so the CJS build should be irrelevant. I think when can keep this as is, but should create an issue to investigate further @jdevcs?

Reply:
https://github.com/web3/web3.js/issues/6108